src/client/Microsoft.Identity.Client/Instance/Region/RegionManager.cs

```diff
@@ -170,7 +170,7 @@ private async Task<RegionInfo> DiscoverAsync(ILoggerAdapter logger, Cancellation
             RegionInfo result = null;
             try
             {
-                await _lockDiscover.WaitAsync().ConfigureAwait(false);
+                await _lockDiscover.WaitAsync(requestCancellationToken).ConfigureAwait(false);
```

Comment:
Should move this cancellable semaphore outside of the try.

https://stackoverflow.com/questions/72983452/canceling-a-semaphoreslim-waitasynccancellationtoken-does-not-always-throw-ope